spring-cloud-dataflow-rest-client/src/main/java/org/springframework/cloud/dataflow/rest/client/TaskOperations.java

```diff
@@ -122,21 +122,24 @@ public interface TaskOperations {
 	TaskExecutionResource taskExecutionStatus(long id);
 
 	/**
-	 * Return the task execution log.  The platform from which to retrieve the log will be set to {@code default}.
+	 * Return the task execution log.
 	 *
 	 * @param externalExecutionId the external execution identifier of the task execution.
+	 * The default is "external" which means the method will use the {@code externalExecutionId}.  If you wish to use the {@code taskExecutionId} created by SCDF then set {@code idType} to "internal".
 	 * @return {@link String} containing the log.
 	 */
 	String taskExecutionLog(String externalExecutionId);
 
 	/**
 	 * Return the task execution log.
 	 *
-	 * @param externalExecutionId the external execution identifier of the task execution.
+	 * @param executionId the external execution identifier of the task execution.
 	 * @param platform the platform from which to obtain the log.
+	 * @param idType identifies whether the log should be obtained using the {@code externalExecutionId} or the {@code taskExecutionId} created by SCDF.
+	 * The default is "external" which means the method will use the {@code externalExecutionId}.  If you wish to use the {@code taskExecutionId} created by SCDF then set {@code idType} to "internal".
 	 * @return {@link String} containing the log.
 	 */
-	String taskExecutionLog(String externalExecutionId, String platform);
```

Comment:
I think we need to leave this existing method in place as well (breaking change otherwise). We can mark it as `@deprecated` and then have the impl just delegate to the new method. Wdyt?